pkg/reconciler/service/occupancytracker.go

```diff
@@ -137,7 +139,9 @@ func (t *OccupancyTracker) AssignCallbackURL(callbackURL string) {
 		t.ticker.Reset(defaultInterval)
 	} else if t.occupancyID != "" {
 		var err error
+		t.Lock()
 		t.occupancyCallbackURL, err = parseOccupancyCallbackURL(callbackURL, t.occupancyID)
```

Comment:
I think this only locks for parsing the callback URL, is this really the lock condition?

Reply:
Here, I am locking the assignment of `t.occupancyCallbackURL` which will be read by the `(*OccupancyTracker).Track()` method (from a different go routine)